logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -255,6 +264,7 @@ public long write(Queueable element) throws IOException {
 
             this.headPage.write(data, seqNum, this.checkpointMaxWrites);
             this.unreadCount++;
+            this.currentSize += data.length;
```

Comment:
That was my initial approach, but if you allocate a page of 10mb and already have 90mb used and the limit is 100mb, does that mean that no writes on the last page will be accepted (since 90+10 is 100 so `isFull()` is `true`?).

The alternative is to only add the page size to counter when the page is full, but that means we'll go over the limit at most 1 page size.